# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_iproute2.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -urN iproute2-2.6.10-ss050112/include/libnetlink.h iproute2-2.6.10-ss050112-diet/include/libnetlink.h
--- iproute2-2.6.10-ss050112/include/libnetlink.h	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/include/libnetlink.h	2005-01-15 00:20:05.355728253 +0100
@@ -1,6 +1,7 @@
 #ifndef __LIBNETLINK_H__
 #define __LIBNETLINK_H__ 1
 
+#include <stdio.h>
 #include <asm/types.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
diff -urN iproute2-2.6.10-ss050112/include/utils.h iproute2-2.6.10-ss050112-diet/include/utils.h
--- iproute2-2.6.10-ss050112/include/utils.h	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/include/utils.h	2005-01-15 00:20:05.355728253 +0100
@@ -1,8 +1,8 @@
 #ifndef __UTILS_H__
 #define __UTILS_H__ 1
 
-#include <asm/types.h>
-#include <resolv.h>
+// #include <asm/types.h>
+// #include <resolv.h>
 
 #include "libnetlink.h"
 #include "ll_map.h"
diff -urN iproute2-2.6.10-ss050112/ip/ipaddress.c iproute2-2.6.10-ss050112-diet/ip/ipaddress.c
--- iproute2-2.6.10-ss050112/ip/ipaddress.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/ipaddress.c	2005-01-15 00:20:05.356728114 +0100
@@ -25,7 +25,7 @@
 #include <string.h>
 #include <fnmatch.h>
 
-#include <linux/netdevice.h>
+// #include <linux/netdevice.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -urN iproute2-2.6.10-ss050112/ip/iplink.c iproute2-2.6.10-ss050112-diet/ip/iplink.c
--- iproute2-2.6.10-ss050112/ip/iplink.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/iplink.c	2005-01-15 00:20:05.357727976 +0100
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff -urN iproute2-2.6.10-ss050112/ip/ipmaddr.c iproute2-2.6.10-ss050112-diet/ip/ipmaddr.c
--- iproute2-2.6.10-ss050112/ip/ipmaddr.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/ipmaddr.c	2005-01-15 00:20:05.358727837 +0100
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff -urN iproute2-2.6.10-ss050112/ip/ipmroute.c iproute2-2.6.10-ss050112-diet/ip/ipmroute.c
--- iproute2-2.6.10-ss050112/ip/ipmroute.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/ipmroute.c	2005-01-15 00:20:05.359727698 +0100
@@ -10,6 +10,9 @@
  *
  */
 
+#define _LINUX_TYPES_H
+#include <asm/types.h>
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff -urN iproute2-2.6.10-ss050112/ip/iptunnel.c iproute2-2.6.10-ss050112-diet/ip/iptunnel.c
--- iproute2-2.6.10-ss050112/ip/iptunnel.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/iptunnel.c	2005-01-15 00:20:05.360727559 +0100
@@ -26,6 +26,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
+#include <asm/byteorder.h>
 #include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/ip.h>
diff -urN iproute2-2.6.10-ss050112/ip/ipxfrm.c iproute2-2.6.10-ss050112-diet/ip/ipxfrm.c
--- iproute2-2.6.10-ss050112/ip/ipxfrm.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/ipxfrm.c	2005-01-15 00:20:05.361727421 +0100
@@ -32,6 +32,7 @@
 #include <sys/socket.h>
 #include <time.h>
 #include <netdb.h>
+#include <netinet/in.h>
 #include <net/if.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
diff -urN iproute2-2.6.10-ss050112/ip/xfrm.h iproute2-2.6.10-ss050112-diet/ip/xfrm.h
--- iproute2-2.6.10-ss050112/ip/xfrm.h	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/ip/xfrm.h	2005-01-15 00:20:05.362727282 +0100
@@ -26,7 +26,7 @@
 #define __XFRM_H__ 1
 
 #include <stdio.h>
-#include <sys/socket.h>
+// #include <sys/socket.h>
 #include <linux/xfrm.h>
 
 #ifndef IPPROTO_SCTP
diff -urN iproute2-2.6.10-ss050112/lib/ll_addr.c iproute2-2.6.10-ss050112-diet/lib/ll_addr.c
--- iproute2-2.6.10-ss050112/lib/ll_addr.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/lib/ll_addr.c	2005-01-15 00:20:05.362727282 +0100
@@ -21,7 +21,7 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
+// #include <linux/netdevice.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -urN iproute2-2.6.10-ss050112/lib/ll_proto.c iproute2-2.6.10-ss050112-diet/lib/ll_proto.c
--- iproute2-2.6.10-ss050112/lib/ll_proto.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/lib/ll_proto.c	2005-01-15 00:20:05.363727143 +0100
@@ -21,7 +21,7 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
+// #include <linux/netdevice.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -urN iproute2-2.6.10-ss050112/lib/ll_types.c iproute2-2.6.10-ss050112-diet/lib/ll_types.c
--- iproute2-2.6.10-ss050112/lib/ll_types.c	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/lib/ll_types.c	2005-01-15 00:20:05.364727004 +0100
@@ -21,7 +21,7 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
+// #include <linux/netdevice.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -urN iproute2-2.6.10-ss050112/Makefile iproute2-2.6.10-ss050112-diet/Makefile
--- iproute2-2.6.10-ss050112/Makefile	2005-01-12 21:19:46.000000000 +0100
+++ iproute2-2.6.10-ss050112-diet/Makefile	2005-01-15 00:20:28.643496189 +0100
@@ -10,7 +10,7 @@
 DEFINES= -DRESOLVE_HOSTNAMES
 
 #options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
-LDLIBS=-lresolv
+LDLIBS=
 ADDLIB=
 
 #options for decnet
@@ -25,7 +25,7 @@
 
 LDLIBS += -L../lib -lnetlink -lutil
 
-SUBDIRS=lib ip tc misc
+SUBDIRS=lib ip
 
 LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
 
